plotLightcurve.py

```diff
@@ -1,152 +1,109 @@
+import diaObjectAnalysis as doa
 
-Two plots are created: one for the light curve with a single set of
-processed/template/difference cutouts, and one showing all of the
-processed/difference cutouts.
+"""Utilities to make cutouts and light curves for DIA Sources
+and DIA Objects using an Alert Production Database (APDB).
 """
 
 
-def main():
-    afwDisplay.setDefaultBackend('matplotlib')
-    script = sys.argv[0]
-    try:
-        repo = sys.argv[1]
-        templateRepo = sys.argv[2]
-    except IndexError:
-        print('###')
-        print('Run this script with 2-3 arguments: repo (required), \
-               templateRepo (required), and dbName (optional)')
-        print('Please note that dbName path must be relative to repo')
-        print('For example, python {0} rerun/MyAmazingRerun rerun/TemplateLand association.db'.format(script))
-        print('###')
-    else:
-        repo = os.path.normpath(repo)
-        rerunName = os.path.basename(repo)
-        # TODO: UN-HARDWIRE THIS
-        miniRegion = 'and ra > 155.2 and ra < 155.3 and decl < -5.6 and decl > -5.8 \
-                      and nDiaSources > 5'  # and flags == 0'
-        try:
-            dbName = sys.argv[3]
-        except IndexError:
-            print('Using default dbName, association.db')
-            dbName = 'association.db'
-        finally:
-            print('Loading APDB Objects...')
-            objTable = loadApdbObjects(repo, dbName, filter=miniRegion)
-            objIdList = list(objTable['diaObjectId'])
-            print('Loaded {0} DIA Objects'.format(len(objIdList)))
-            # if len(objIdList) > 20:
-            #     objIdList = objIdList[0:20]
-            #     print('WARNING: objIdList has been truncated!')
-            print('Plotting light curves for {0} objects...'.format(len(objIdList)))
-            patchList = ['11,8', '11,9', '11,10']  # TODO: UN-HARDWIRE THIS
-            pdfLightcurves = PdfPages(rerunName + '_lcs.pdf')
-            pdfCutouts = PdfPages(rerunName + '_cos.pdf')
-            for obj in objIdList:
-                plotLightcurve(obj, objTable, repo, dbName, templateRepo,
-                               patchList, plotAllCutouts=True,
-                               pdfLc=pdfLightcurves, pdfCo=pdfCutouts)
-            pdfLightcurves.close()
-            pdfCutouts.close()
-
-
-def in_ipynb():
-    """Determine whether the code is being run in a notebook.
-    """
-    try:
-        cfg = get_ipython().config
-        if cfg['IPKernelApp']['parent_appname'] == 'ipython-notebook':
-            return True
-        elif 'LazyConfigValue' in str(cfg['IPKernelApp']['parent_appname']):
-            return True
-        else:
-            return False
-    except NameError:
-        return False
-
-
-def loadApdbObjects(repo, dbName='association.db', filter=''):
-    """Load select DIAObject columns from a APDB into a pandas dataframe.
+def loadSomeApdbSources(dbName, diaObjectId, dbType='sqlite', schema=None):
```

Comment:
Could this be called `loadApdbSources` or `loadSelectedApdbSources`? `Some` sounds like they could be random.

Reply:
`loadAnyOldApdbSources` ... haha fair enough